docs_2.0/manual_EN/5.deployment-and-installation/2.compile-and-install-nebula-graph/1.install-nebula-graph-by-compiling-the-source-code.md

```diff
@@ -0,0 +1,128 @@
+
+    ```bash
+    $ make -j{N}
+    ```
+
+6. Install Nebula Graph.
+
+    ```bash
+    $ make install-all
+    ```
+
+## CMake variables
+
+Usage of CMake variables:
+
+```bash
+$ cmake -DVariable=<value> ...
+```
+
+The following CMake variables can be used at the configure (cmake) stage to adjust the compiling settings.
+
+### ENABLE_BUILD_STORAGE
+
+Starting from the 2.0 pre-release, Nebula Graph supports separated compute and storage. The `ENABLE_BUILD_STORAGE` variable is set to `OFF` by default so that the storage service is not installed together with the graph service.
+
+If you are deploying Nebula Graph on a single host for testing, you can set `ENABLE_BUILD_STORAGE` to `ON` to download and install the storage service automatically.
+
+### CMAKE_INSTALL_PREFIX
+
+`CMAKE_INSTALL_PREFIX` specifies the path where the service modules, scripts, configuration files are installed. The default path is `/usr/local/nebula`.
+
+### ENABLE_WERROR
+
+`ENABLE_WERROR` is `ON` by default and it makes all warnings into errors. You can set it to `OFF` if needed.
+
+### ENABLE_TESTING
+
+`ENABLE_TESTING` is `ON` by default and unit tests are built with the Nebula Graph services. If you just need the service modules, set it to `OFF`.
+
+### ENABLE_ASAN
+
+`ENABLE_ASAN` is `OFF` by default and the building of ASan (AddressSanitizer), a memory error detector, is disabled. To enable it, set `ENABLE_ASAN` to `ON`. This variable is intended for Nebula Graph developers.
+
+### CMAKE_BUILD_TYPE
+
+Nebula Graph supports the following building types:
+
+* `Debug`, the default value of `CMAKE_BUILD_TYPE`, indicates building Nebula Graph with the debug info but not the optimization options.
+* `Release`, indicates building Nebula Graph with the optimization options but not the debug info.
+* `RelWithDebInfo`, indicates building Nebula Graph with the optimization options and the debug info.
+* `MinSizeRel`, indicates building Nebula Graph with the optimization options for controlling the code size but not the debug info.
+
+### CMAKE_C_COMPILER/CMAKE_CXX_COMPILER
+
+Usually, CMake locates and uses a C/C++ compiler installed in the host automatically. But if your compiler is not installed at the standard path, or if you want to use a different one, run the command as follows to specify the installation path of the target compiler:
+
+```bash
+$ cmake -DCMAKE_C_COMPILER=/path/to/gcc/bin/gcc -DCMAKE_CXX_COMPILER=/path/to/gcc/bin/g++ ..
+$ cmake -DCMAKE_C_COMPILER=/path/to/clang/bin/clang -DCMAKE_CXX_COMPILER=/path/to/clang/bin/clang++ ..
+```
+
+### ENABLE_CCACHE
+
+`ENABLE_CCACHE` is `ON` by default and ccache is used to speed up the compiling of Nebula Graph.
+
+To disable `ccache`, set `ENABLE_CCACHE` to `OFF`. On some platforms, the `ccache` installation hooks up or precedes the compiler. In such a case, you have to set an environment variable `export CCACHE_DISABLE=true` or add a line `disable=true` in `~/.ccache/ccache.conf` as well.
```

Comment:
In such a case, you have to set an environment variable `export CCACHE_DISABLE=true` or add a line `disable=true` in `~/.ccache/ccache.conf` as well. ==> In such a case, you must set `export CCACHE_DISABLE=true`, or in the `~/.ccache/ccache.conf` file, add a line `disable=true`.